rustls/src/lib.rs

```diff
@@ -409,7 +409,7 @@ pub use crate::tls12::Tls12CipherSuite;
 pub use crate::tls13::Tls13CipherSuite;
 pub use crate::verify::DigitallySignedStruct;
 pub use crate::versions::{SupportedProtocolVersion, ALL_VERSIONS, DEFAULT_VERSIONS};
-pub use crate::webpki::{RootCertStore, TrustAnchorWithDn};
+pub use crate::webpki::{RootCertStore, TrustAnchorWithDn, WebPkiSupportedAlgorithms};
```

Comment:
This looks kinda confusing to me. As far as I can tell, all of these types are independent of crypto providers yet they live within the `webpki` module, and the `WebPkiSupportedAlgorithms` even carries the `WebPki` prefix. How does that make sense?

Reply:
I think these are all implementation details of `WebPki{Server,Client}Verifier` and so aren't relevant unless webpki is being used:

- `TrustAnchorWithDn`: needed for `WebPkiClientVerifier` to implement `ClientCertVerifier::client_auth_root_subjects`. This is mainly a polyfill over internal details pki-types/webpki's `TrustAnchor` encoding. I wonder if we should change that to avoid the need for this type?
- `WebPkiSupportedAlgorithms` is an implementation detail for `{Client,Server}CertVerifier::supported_verify_schemes`, but mostly the impedence mismatch between TLS1.2 and pki-types/webpki `SignatureVerificationAlgorithm`. I think that mismatch is OK, and feel the change to TLS1.3 which removes it supports that viewpoint.
- `RootCertStore` is really hiding implementation details of webpki -- that it needs an extra step to deal with a trust anchor available as DER.

Reply:
See #1459 for some steps to simplify this stuff.